cabal-install/Main.hs

```diff
@@ -268,6 +270,9 @@ mainWorker args = topHandler $
       , hiddenCmd  win32SelfUpgradeCommand win32SelfUpgradeAction
       , hiddenCmd  actAsSetupCommand actAsSetupAction
       , hiddenCmd  manpageCommand (manpageAction commandSpecs)
+
+      , hiddenCmd  installCommand { commandName = "new-configure" } newConfigureAction
```

Comment:
Not really. Currently configure just does the first phases of build, and currently the planning phase will download packages. Eventually configure will persist config flags by writing them info a local file that extends the cabal.project file.

Note that once we have the hackage security stuff merged then the planning phase will not need to download tarballs, since it'll have the tarball hashes up front.